Sources/News.php

```diff
@@ -164,55 +200,44 @@ function ShowXmlFeed()
 		list ($total_posts) = $smcFunc['db_fetch_row']($request);
 		$smcFunc['db_free_result']($request);
 
-		$feed_meta['title'] = ' - ' . strip_tags($board_info['name']);
+		$feed_meta['title'] = $board_info['name'];
 		$feed_meta['source'] .= '?board=' . $board . '.0';
 
 		$query_this_board = 'b.id_board = ' . $board;
 
 		// Try to look through just a few messages, if at all possible.
 		if ($total_posts > 80 && $total_posts > $modSettings['totalMessages'] / 10)
-			$context['optimize_msg']['lowest'] = 'm.id_msg >= ' . max(0, $modSettings['maxMsgID'] - 600 - $_GET['limit'] * 5);
+			$context['optimize_msg']['lowest'] = 'm.id_msg >= ' . max(0, $modSettings['maxMsgID'] - 600 - $context['xmlnews_limit'] * 5);
 	}
 	else
 	{
 		$query_this_board = '{query_see_board}' . (!empty($modSettings['recycle_enable']) && $modSettings['recycle_board'] > 0 ? '
 			AND b.id_board != ' . $modSettings['recycle_board'] : '');
-		$context['optimize_msg']['lowest'] = 'm.id_msg >= ' . max(0, $modSettings['maxMsgID'] - 100 - $_GET['limit'] * 5);
+		$context['optimize_msg']['lowest'] = 'm.id_msg >= ' . max(0, $modSettings['maxMsgID'] - 100 - $context['xmlnews_limit'] * 5);
 	}
 
-	// Show in rss or proprietary format?
-	$xml_format = isset($_GET['type']) && in_array($_GET['type'], array('smf', 'rss', 'rss2', 'atom', 'rdf')) ? $_GET['type'] : 'rss2';
-
-	// @todo Birthdays?
-
-	// List all the different types of data they can pull.
-	$subActions = array(
-		'recent' => array('getXmlRecent', 'recent-post'),
-		'news' => array('getXmlNews', 'article'),
-		'members' => array('getXmlMembers', 'member'),
-		'profile' => array('getXmlProfile', null),
-	);
-
-	// Easy adding of sub actions
-	call_integration_hook('integrate_xmlfeeds', array(&$subActions));
+	$feed_meta['title'] .= (!empty($feed_meta['title']) ? ' - ' : '') . $context['forum_name'];
 
-	if (empty($_GET['sa']) || !isset($subActions[$_GET['sa']]))
-		$_GET['sa'] = 'recent';
+	// Sanitize feed metadata values
+	foreach ($feed_meta as $mkey => $mvalue)
+		$feed_meta[$mkey] = strip_tags($mvalue);
 
 	// We only want some information, not all of it.
-	$cachekey = array($xml_format, $_GET['action'], $_GET['limit'], $_GET['sa']);
+	$cachekey = array($xml_format, $_GET['action'], $context['xmlnews_limit'], $subaction, $context['xmlnews_offset']);
 	foreach (array('board', 'boards', 'c') as $var)
-		if (isset($_REQUEST[$var]))
-			$cachekey[] = $_REQUEST[$var];
+		if (isset($_GET[$var]))
+			$cachekey[] = $var . '=' . implode(',', $_GET[$var]);
 	$cachekey = md5($smcFunc['json_encode']($cachekey) . (!empty($query_this_board) ? $query_this_board : ''));
 	$cache_t = microtime(true);
 
 	// Get the associative array representing the xml.
 	if (!empty($cache_enable) && (!$user_info['is_guest'] || $cache_enable >= 3))
```

Comment:
Missing global

Reply:
Nice catch. It looks like that bug has been lurking unnoticed for several years. I'll toss in a fix.